src/Entity/ShippingMethodInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Shipping Slot plugin for Sylius.
+ *
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusShippingSlotPlugin\Entity;
+
+use Sylius\Component\Core\Model\ShippingMethodInterface as SyliusShippingMethodInterface;
+
+interface ShippingMethodInterface extends SyliusShippingMethodInterface
```

Comment:
You should not extend the main interface, the shipping method object will do it for you!

Reply:
Yes but for IDE autocomplete we will have the interface used in the code and we will not have Sylius ShippingMethodInterface methods

Reply:
You can implement more than one interface on an object.